### PR TITLE
NOI Decisions Pt 1

### PR DESCRIPTION
**Pull request by dhaselhan**

### CODE REVIEW

File: alcs-frontend/src/app/services/notice-of-intent/decision/notice-of-intent-decision.service.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+      await firstValueFrom(this.http.delete<NoticeOfIntentDecisionDto>(`${this.url}/${uuid}`));
+      this.toastService.showSuccessToast('Decision deleted');
+    } catch (err) {
+      this.toastService.showErrorToast('Failed to delete decision');
+    }
+  }
+
+  async uploadFile(decisionUuid: string, file: File) {
+    if (file.size > environment.maxFileSize) {
+      const niceSize = environment.maxFileSize / 1048576;
```
Comment (mhuseinov) [inline alcs-frontend/src/app/services/notice-of-intent/decision/notice-of-intent-decision.service.ts:92]: I guess it is good time to move this '1048576' into some kind of constant

File: services/apps/alcs/src/alcs/application-decision/application-decision-outcome.entity.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { Column, Entity } from 'typeorm';
+import { BaseCodeEntity } from '../../common/entities/base.code.entity';
+
+@Entity()
+export class ApplicationDecisionOutcomeCode extends BaseCodeEntity {
+  @Column({
+    default: true,
+  })
+  isFirstDecision: boolean;
```
Comment (mhuseinov) [inline services/apps/alcs/src/alcs/application-decision/application-decision-outcome.entity.ts:9]: So this one changed in scope of NOI and isFirst will stored on db level for both decision v1 and v2?

Comment (dhaselhan) [reply services/apps/alcs/src/alcs/application-decision/application-decision-outcome.entity.ts:9]: I did not change this file other than renaming it and moving it.

File: services/apps/alcs/src/alcs/notice-of-intent-decision/notice-of-intent-decision.entity.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+  @AutoMap(() => NoticeOfIntentDecisionOutcome)
+  @ManyToOne(() => NoticeOfIntentDecisionOutcome, {
+    nullable: false,
+  })
+  outcome: NoticeOfIntentDecisionOutcome;
+
+  @AutoMap()
+  @Column()
+  outcomeCode: string;
```
Comment (mhuseinov) [inline services/apps/alcs/src/alcs/notice-of-intent-decision/notice-of-intent-decision.entity.ts:44]: no column descriptions 🥲

File: services/apps/alcs/src/providers/typeorm/migrations/1685640625670-prefix_application_decision_tables.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class prefixApplicationDecisionTables1685640625670
+  implements MigrationInterface
+{
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `ALTER TABLE "alcs"."decision_document" RENAME TO "application_decision_document"`,
```
Comment (mhuseinov) [inline services/apps/alcs/src/providers/typeorm/migrations/1685640625670-prefix_application_decision_tables.ts:8]: I wonder if this affects anything on metabase

**Review by mhuseinov — APPROVED** [review]
🔥

File: services/apps/alcs/src/alcs/notice-of-intent-decision/notice-of-intent-decision.entity.ts (hunk 3 above)
Comment (dhaselhan) [reply services/apps/alcs/src/alcs/notice-of-intent-decision/notice-of-intent-decision.entity.ts:44]: I have a feeling we won't be going back and adding them anytime soon

File: alcs-frontend/src/app/services/notice-of-intent/decision/notice-of-intent-decision.service.ts (hunk 1 above)
Comment (dhaselhan) [reply alcs-frontend/src/app/services/notice-of-intent/decision/notice-of-intent-decision.service.ts:92]: Created a file size checker that has a clearly defined constant and replaced all usages (other than Portal) with that

**Pull request merged**